app/modules/account.js

```diff
@@ -18,8 +19,15 @@ export function logout(){
   }
 };
 
+export function showKey(isToggled) {
```

Comment:
So to have component state you need to change it first from a function to a class that extends the React component class. Then we can have self contained states, like in the example below.

Feel free to use this but I didn't test it.

```
class Login extends Component {
    static propTypes = {
        loggedIn: PropTypes.bool.isRequired,
        wif: PropTypes.string.isRequired,
    };

    state = {
        showKey: false
    };

    toggleKeyVisibility = () => {
        this.setState(prevState => ({
            showKey: !prevState.showKey
        }));
    };
    
    handleInputChange = e => {
        this.setState({
            [e.target.name]: e.target.value
        });
    };

    render = () => {
        const { loggedIn, wif } = this.props;
        const { showKey } = this.state;

        return (
            <div id="loginPage">
                <div className="login">
                    <div className="logo">
                        <img src={logo} width="60px" />
                    </div>

                    <input type={showKey ? 'text' : 'password'}
                           placeholder="Enter your private key here (WIF)"
                           id="inputKey"
                           name="wif"
                           onChange={this.handleInputChange} />

                    {showKey ?
                        <FaEyeSlash className="viewKey" onClick={this.toggleKeyVisibility} /> :
                        <FaEye className="viewKey" onClick={this.toggleKeyVisibility} />}

                    <div className="loginButtons">
                        {loggedIn ?
                            <Link to="/dashboard">
                                <button>Login</button>
                            </Link> :
                            <button disabled="true">Login</button>}
                        <Link to="/create">
                            <button>New Wallet</button>
                        </Link>
                    </div>
                </div>
            </div>
        )
    }
}

const mapStateToProps = state => ({
    loggedIn: state.account.loggedIn,
    wif: state.account.wif,
});

const mapActionCreators = (dispatch) => {
    onWifChange: (dispatch, value) => {
        // TODO: changed back to only WIF login for now, getting weird errors with private key hex login
        dispatch(login(value));
    }
};

export default connect(mapStateToProps, mapActionCreators)(Login);
```